hooks/contexts/useErrorContext.ts

```diff
@@ -1,4 +1,13 @@
-import { ErrorContext } from "contexts";
+import { DefaultError, PanelError } from "contexts";
 import { useContext } from "react";
 
-export const useErrorContext = () => useContext(ErrorContext);
+export const useErrorContext = (type?:string) => {
```

Comment:
we can add arbitrary new error contexts here.  @ryanwolhuter im not sure if i should be dynamically calling useContext like this though, rather than having a static variable

Reply:
maybe it will be simpler to just have a `useErrorContext` and `usePanelErrorContext` to avoid this? I think this is cool but I don't know what the implications might be down the road.

Reply:
good idea